tools/roslaunch/src/roslaunch/xmlloader.py

```diff
@@ -301,6 +302,9 @@ def _arg_tag(self, tag, context, ros_config, verbose=True):
         except substitution_args.ArgException as e:
             raise XmlParseException(
                 "arg '%s' is not defined. \n\nArg xml is %s"%(e, tag.toxml()))
+        except ResourceNotFound as e:
+            raise XmlParseException(
+                "Invalid <arg> tag: Make sure the following is found in ROS_PACKAGE_PATH: %s \n\nArg xml is %s"%(e, tag.toxml()))
```

Comment:
In this case the tag itself is not "invalid". Also this is not a problem "parsing the XML" so the exception type is confusing.

Reply:
That makes sense. Please see the update.